src/syngen/ml/processors/processors.py

```diff
@@ -0,0 +1,368 @@
+        """
+        json_columns = list()
+        for column in data.columns.to_list():
+            try:
+                if pd.isnull(data[column]).all():
+                    continue
+                data[column].dropna().apply(lambda v: json.loads(v))
+                json_columns.append(column)
+            except (TypeError, JSONDecodeError):
+                continue
+        return json_columns
+
+    @staticmethod
+    def _get_flattened_df(
+            data: pd.DataFrame,
+            json_columns: List,
+    ) -> Tuple[pd.DataFrame, Dict, List]:
+        """
+        Flatten the JSON columns in the dataframe
+        """
+        df_list = list()
+        flattening_mapping = dict()
+        for column in json_columns:
+            flattened_data = pd.DataFrame(
+                [
+                    flatten(json.loads(i), ".")
+                    for i in data[column]
+                ]
+            )
+            flattening_mapping[column] = flattened_data.columns.to_list()
+            df_list.append(flattened_data)
+        flattened_data = pd.concat([data, *df_list], axis=1)
+        duplicated_columns = [
+            key
+            for key, value in dict(Counter(flattened_data.columns.to_list())).items()
+            if value > 1
+        ]
+        flattened_data.drop(columns=flattening_mapping.keys(), inplace=True)
+        flattened_df = flattened_data.T.loc[~flattened_data.T.index.duplicated(), :].T
+        flattened_df = flattened_df.applymap(lambda x: np.NaN if x in [list(), dict()] else x)
+        return flattened_df, flattening_mapping, duplicated_columns
+
+    def _save_flatten_metadata(self, metadata: Dict):
+        """
+        Save the metadata of the flattening process
+        """
+        with open(f"{self.path_to_flatten_metadata}", "w") as f:
+            json.dump(metadata, f)
+
+    @staticmethod
+    def _load_source(source: str, *args):
+        """
+        Load the data from the predefined source
+        """
+        return DataLoader(source).load_data()
+
+    @staticmethod
+    def _save_input_data(
+            path_to_input_data: str,
+            flattened_data: pd.DataFrame,
+            *args
+    ):
+        """
+        Save the input data to the predefined path
+        """
+        DataLoader(path_to_input_data).save_data(path_to_input_data, flattened_data)
+
+    def _handle_json_columns(self):
+        """
+        Preprocess the data contained JSON columns before the training process
+        """
+        metadata = self._fetch_metadata()
+        flatten_metadata = dict()
+        for table, settings in metadata.items():
+            if table == "global":
+                continue
+            path_to_input_data = (f"{PATH_TO_MODEL_ARTIFACTS}/tmp_store/{slugify(table)}/"
+                                  f"input_data_{slugify(table)}.pkl")
+            source = settings.get("train_settings", {}).get("source", "")
+            data, schema = self._load_source(source, table)
```

Comment:
It seems that argument table is redundant.

Reply:
It's not redundant. It sends to the function for the Enterprise version